titiler/api/utils.py

```diff
@@ -17,7 +17,7 @@
 
 def get_cache(request: Request) -> CacheLayer:
     """Get Memcached Layer."""
-    return request.state.cache
+    return getattr(request.state, "cache", None)
```

Comment:
@bitner cache is always set here https://github.com/developmentseed/titiler/blob/725da5fa1bc56d8e192ae8ff0ad107493ca93378/titiler/main.py#L55 so it should not error IMO.

I'm fine with adding a safeguard in case someone wants to use titiler module without the main app. 

Note:  I think `state` will behave like a dict so I'll prefer if we do `request.state.get("cache", None)`

Reply:
We did run into this problem when using this as a module without the cache middleware.